core/src/main/scala/tech/beshu/ror/boot/engines/BaseReloadableEngine.scala

```
@@ -102,6 +102,7 @@ private[engines] abstract class BaseReloadableEngine(val name: String,
       _ <- canBeReloaded(newConfig)
       newEngineWithConfig <- reloadWith(newConfig, newConfigEngineTtl)
       _ <- replaceCurrentEngine(newEngineWithConfig)
```

I noticed that it can return None in Right. It looks like we could throw an Invalid State exception when the result state is NotStartedYet and always return EngineWithConfig in Right

---

In `replaceCurrentEngine` method `getAndTransform` was called so result was the old engine. 
The old enginge is closed in transform part so we could ignore result and return just Unit type.